docs/guides/publishing.md

```diff
@@ -5,52 +5,134 @@ description: Publishing A Component Library
+To ensure that the right entry file is loaded when importing the project, define the following fields in your `package.json`:
 
 ```json
 {
-  "main": "dist/index.cjs.js",
-  "module": "dist/index.js",
-  "es2015": "dist/esm/index.mjs",
-  "es2017": "dist/esm/index.mjs",
-  "types": "dist/types/components.d.ts",
-  "unpkg": "dist/my-project-name/my-project-name.esm.js",
-  "collection:main": "dist/collection/index.js",
-  "collection": "dist/collection/collection-manifest.json",
-  "files": [
-    "dist/",
-    "css/",
-    "loader/"
-  ]
+  "exports": "./dist/esm/my-design-system.js",
+  "main": "./dist/cjs/my-design-system.js",
+  "unpkg": "dist/my-design-system/my-design-system.esm.js",
+}
+```
+
+Read more about various options when it comes to configuring your project's components for lazy loading in the [`dist`](../output-targets/dist.md) output target section.
+
+#### Considerations
+
+To start, Stencil was designed to lazy-load itself only when the component was actually used on a page. There are many benefits to this approach, such as simply adding a script tag to any page and the entire library is available for use, yet only the components actually used are downloaded. For example, [`@ionic/core`](https://www.npmjs.com/package/@ionic/core) comes with over 100 components, but a webpage may only need `ion-toggle`. Instead of requesting the entire component library, or generating a custom bundle for just `ion-toggle`, the `dist` output target is able to generate a tiny entry build ready to load any of its components on-demand.
+
+However be aware that this approach is not ideal in all cases. It requires your application to ship the bundled components as static assets in order for them to load properly. Furthermore, having many nested component dependencies can have an impact on the performance of your application. For example, given you have a component `CmpA` which uses a Stencil component `CmpB` which itself uses another Stencil component `CmpC`. In order to fully render `CmpA` the browser has to load 3 scripts sequentially which can result in undesired rendering delays.
+
+### Standalone
+
+The [`dist-custom-elements`](../output-targets/custom-elements.md) output target builds each component as a stand-alone class that extends `HTMLElement`. The output is a standardized custom element with the styles already attached and without any of Stencil's lazy-loading. This may be preferred for projects that are already handling bundling, lazy-loading and defining the custom elements themselves.
+
+You can use these standalone components by importing them via:
+
+```ts
+import { MyComponent, defineCustomElementMyComponent } from 'my-design-system'
+
+// register to CustomElementRegistry
+defineCustomElementMyComponent()
+
+// or extend custom element via
+class MyCustomComponent extends MyComponent {
+  // ...
```

Comment:
Yes, for example, let's say we publish a `create-stencil` component, you can change functionality by extending the class and overwriting component methods, e.g.:

```html
<script type="module">
  import { MyComponent } from 'https://unpkg.com/foobar-design-system@0.0.4/dist/components/index.js'
  class ChangedComponent extends MyComponent {
    getText() {
      return 'a new component!';
    }
  }

  customElements.define('my-component', ChangedComponent);
  customElements.define('my-component-orig', MyComponent);
</script>
<my-component first="foo" last="bar"></my-component>
<my-component-orig first="foo" last="bar"></my-component-orig>
```

This will result in:

<img width="259" alt="Screenshot 2024-02-28 at 10 16 25 AM" src="https://github.com/ionic-team/stencil-site/assets/731337/3120ab4b-998f-4afd-879d-5b02eaec35c6">

Reply:
wow I didn't know that worked! I guess I hadn't tried it

Reply:
Yeah extending the compiled class output will work, but obviously wouldn't be able to add any additional reactivity (states, props, watchers, etc.) at this time.